fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -2548,4 +2641,220 @@ public boolean isContinueIfWriteOnly() {
             return continueIfWriteOnly;
         }
     }
+
+    @Nonnull
+    private CompletableFuture<Void> buildIndexFromIndexMarkBuilt(Subspace subspace) {
+        // zz 2.1 - mark built
+        // Grand Finale - after fully building the index, remove all missing ranges in one gulp
+        // Note that we do not assume srcIndex has the same order as the record store, therefore did not try to handle
+        // missing spaces (for now..)
```

Comment:
My point was that would we assume the same order, we could have updated the rangeSet (by record, not by index) and - potentially - allow a record-scan continue an index-scan. But since it is indeed misleading, I removed the comment.